### ISSUE TITLE
graph for iostat wait time:  axis label doesn't match "value" column in table alongside

### ISSUE THREAD

**Issue by bengland2** [distributed-system-analysis/pbench#412]
This is really important. In [this graph](http://abbynormal.rdu.openstack.engineering.redhat.com/ceph-tests/cbt-librbdfio-2016-10-23-11-38-44/1/reference-result/tools-default/c04-h29-6048r/iostat/disk.html) you can see the disparity between the y-axis label, which reaches up to 1800 (msec) and the peak "value" column (when I hold the mouse right over the highest point in the graph).   Am I correct that there is a mismatch and if so what can be done about it?

**Comment by k-rister** [comment]
It seems right to me:

![foo](https://cloud.githubusercontent.com/assets/15911116/19657708/ec396e9a-99ea-11e6-8e04-17cda19a91bc.png)

Is this not what you see?

**Comment by bengland2** [comment]
You are correct, it is there, I just had trouble seeing it.  So there were so many disks x {write,read} curves in this plot that I didn't see the value I was hovering over, I looked harder and way down at the bottom of the 2nd page was the thing I was looking for, because I use big fonts.   It would be great if it sorted by the "value" column when you hover over a point on the graph, then it would show the highest value at that point up top.  But the graph and table is not incorrect, it's just a little hard to read.

**Comment by bengland2** [comment]
Karl, I like your [fix described in e-mail](http://perf1.perf.lab.eng.bos.redhat.com/krister/code/jschart/pbench/disk2.html) to sort based on "value" column when mouse is hovering over a particular data point.  I didn't notice any problems with it.  The table churns a lot as you move the mouse over the graph, so perhaps wait until the mouse isn't moving for 1/4 sec before filling in the "value" column, if it's easy to do?  But this is optional.

**Comment by k-rister** [comment]
Here is a demo of the prototyped solution that Ben just referred to:

[https://k-rister.github.io/pbench/issues/412/001/disk2.html](https://k-rister.github.io/pbench/issues/412/001/disk2.html)

I'm curious what people think of it.  On one hand I think it does exactly what he wants, on the other it is a bit "busy" for lack of a better word as the table is re-sorting itself as the cursor moves around the viewport.

Do people like this or think it's too much action?

I also prototyped the ability to "lock" the cursor (and it's associated activity) by right clicking in the viewport which should also help with this.  Right click again to unlock.  This should have probably already been done, but better late than never.  There are still some rough edges to be ironed out here such as zooming while it's locked.

**Comment by k-rister** [comment]
Ok, I have made a lot of progress on getting these changes working in all facets of the charting library.  It turns out the "lock" the cursor feature required quite a few more changes than I had originally expected to cover all the corner cases.

Here is an updated prototype:

[https://k-rister.github.io/pbench/issues/412/002/disk2.html](https://k-rister.github.io/pbench/issues/412/002/disk2.html)

At this point the action is all in realtime, nothing is deferred until the action stops (such as when the mouse stops moving).  Right now all events are handled as they fire, I feel like deferring event handling is probably a broader subject that should be addressed separately, if at all.  I'll have to do some research to determine if/how that would work.

On a separate note, I've uncovered a separate issue that I'll be filing another issue on.  Put simply, I believe that when hiding a dataset in a stacked chart that is zoomed the axis scales get messed up until "Reset Zoom/Pan" is clicked.  Sorta odd that neither myself or anyone else has hit this yet, but so it goes.

**Comment by bengland2** [comment]
I think the original issue is resolved, thanks!
